client.go

```diff
@@ -50,28 +50,41 @@ func (c *client) DialBack(ctx context.Context, p peer.ID) (ma.Multiaddr, error)
 	err = w.WriteMsg(req)
 	if err != nil {
 		s.Reset()
-		return nil, err
+		return nil, nil, err
 	}
 
 	var res pb.Message
 	err = r.ReadMsg(&res)
 	if err != nil {
 		s.Reset()
-		return nil, err
+		return nil, nil, err
 	}
 
 	if res.GetType() != pb.Message_DIAL_RESPONSE {
-		return nil, fmt.Errorf("Unexpected response: %s", res.GetType().String())
+		return nil, nil, fmt.Errorf("Unexpected response: %s", res.GetType().String())
 	}
 
 	status := res.GetDialResponse().GetStatus()
 	switch status {
 	case pb.Message_OK:
-		addr := res.GetDialResponse().GetAddr()
-		return ma.NewMultiaddrBytes(addr)
+		res := res.GetDialResponse()
+		success = make([]ma.Multiaddr, 0, len(res.SuccessAddrs))
+		for _, ab := range res.SuccessAddrs {
```

Comment:
do we have any way to check these addresses are plausibly our own? does this mean someone we ask to dial us can give us a list of thousands of addresses back and either cause us to hold on to a lot of bogus addresses or do work?

Reply:
We should get confirmation from multiple peers. We'll need to shift from confirming that we're dialable with some confidence to confirming that some set of addresses are dialable with confidence.

Reply:
Ah, I see that's planned in the next patch. That'll make this easier to review.